pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -104,8 +121,25 @@ def get_batch_log_metrics(self, *args, **kwargs):
     def run_epoch_func(self, results, opt_metric, func_name, *args, **kwargs) -> None:
         if not isinstance(opt_metric, Result):
             raise Exception("The provided opt_metric should be a Result Object. Something is wrong")
+
         func = getattr(opt_metric, func_name)
         metrics_to_log = func(*args, add_dataloader_idx=self.has_several_dataloaders, **kwargs)
+
+        if torch.distributed.is_initialized():
+            device = self._all_gather_fn.__self__.device
+            for non_metric_key in opt_metric.get_non_metrics_keys():
+                if non_metric_key in metrics_to_log and non_metric_key not in warning_cache.warned_metrics:
+                    metric = self._all_gather_fn(metrics_to_log[non_metric_key].to(device))
+                    if any(metric[0] != m for m in metric[1:]):
```

Comment:
Can it happen that metric has only one element? I believe you can initialize distributed with one process

Reply:
Not sure this is possible, but wouldn't the list be empty in that case? And any on an empty list should be false.